core/lib/via_btc_client/src/signer/mod.rs

```diff
@@ -1 +1,360 @@
+use async_trait::async_trait;
+use bitcoin::{
+    hashes::Hash,
+    secp256k1::{Message, Secp256k1},
+    sighash::{Prevouts, SighashCache},
+    taproot::{ControlBlock, LeafVersion},
+    EcdsaSighashType, PrivateKey, ScriptBuf, TapLeafHash, TapSighashType, Transaction, TxOut,
+    Witness,
+};
+use secp256k1::{Keypair, SecretKey};
 
+use crate::{
+    traits::{BitcoinRpc, BitcoinSigner},
+    types::{BitcoinError, BitcoinSignerResult},
+};
+
+pub struct BasicSigner<'a> {
+    private_key: PrivateKey,
+    rpc_client: &'a dyn BitcoinRpc,
```

Comment:
why signer needs, `rpc_client` in its state?

Reply:
we need to use `get_transaction` method to get actual transactions from input.previous_output.txid ([OutPoint](https://docs.rs/bitcoin/0.32.2/bitcoin/struct.OutPoint.html) struct, from input of tx)

Reply:
it just a reference, so that's totally ok as for me